EXTI/Core/Inc/global.h

```diff
@@ -0,0 +1,17 @@
+/*
+ * global.h
+ *
+ *  Created on: Mar 5, 2020
+ *      Author: rodrigocelmer
+ */
+
+#ifndef INC_GLOBAL_H_
+#define INC_GLOBAL_H_
+
+//--- Global Variables
+int user_bt_count	=0,	//counter to debounce the button
+	flag_EXTI		=0,	//flag for EXTI interrupt
+	flag_TIM6		=0,	//flag for TIM6 interrupt
+	flag_frequency	=1;	//flag to indicate if we are increasing or decreasing the frequency
+
```

Comment:
As the flag variable indicates only two values, it needs only one bit to work. Meaning that by declaring it as an int var, it will occupy 32bits in memory. Instead, you can use uint8_t, that occupies only 8 bits.

In C++ exists a bool type, that occupies only one bit, but this type does not exist in C :(